test/test_jit_forms.py

```diff
@@ -1074,3 +1074,73 @@ def test_integral_grouping(compile_args):
         ]
     )
     assert len(unique_integrals) == 2
+
+
+@pytest.mark.parametrize("dtype", ["float64"])
+@pytest.mark.parametrize("permutation", [[0, 0], [0, 1], [1, 0], [1, 1]])
+def test_mixed_dim_form(compile_args, dtype, permutation):
+    # TODO Test 3D and non-simplex
+
+    k = 1
+    ele_type = "Lagrange"
+    V_cell_type = "triangle"
+    Vbar_cell_type = "interval"
+
+    V_ele = basix.ufl.element(ele_type, V_cell_type, k)
+    Vbar_ele = basix.ufl.element(ele_type, Vbar_cell_type, k)
+
+    gdim = 2
+    V_domain = ufl.Mesh(basix.ufl.element("Lagrange", V_cell_type, 1, shape=(gdim,)))
+    Vbar_domain = ufl.Mesh(basix.ufl.element("Lagrange", Vbar_cell_type, 1, shape=(gdim,)))
+
+    V = ufl.FunctionSpace(V_domain, V_ele)
+    Vbar = ufl.FunctionSpace(Vbar_domain, Vbar_ele)
+
+    u = ufl.TrialFunction(V)
+    vbar = ufl.TestFunction(Vbar)
+
+    f = ufl.Coefficient(V)
+    g = ufl.Coefficient(Vbar)
+
+    ds = ufl.Measure("ds", domain=V_domain)
+
+    forms = [ufl.inner(f * g * u, vbar) * ds]
```

Comment:
I think it would be good to test expressions with gradients on coefficients from either mesh, to ensure that FacetJacobians, FacetGradients etc are computed correctly (jorgensd agrees)

Reply:
The test now uses `grad(u) * n`